cci_nepal/analysis/wp2_data_scoping/database_sample/nepal_flood_landslide_descriptive.py

```diff
@@ -0,0 +1,1006 @@
+
+# %%
+# replace the null values for the age groups for onward processing. The assumption here is that missing values
+# imply situations of non-applicability. For example, the absence of male0_5 means no children of this age in houshold
+# columns[22:30] of current dataframe hold data on the composition of households by age
+data_df[data_df.columns[22:30]] = data_df[data_df.columns[22:30]].fillna(0, axis=1)
+
+# %%
+# male/female populations by age groups for the different households.
+beneficiaries_by_age_group = {}
+for col in data_df.columns[22:30]:
+    beneficiaries_by_age_group[col] = sum(data_df[col])
+
+# %%
+# stacked bar chart representing distribution of household members by age groups.
+labels = ["Age0_5", "Age6_17", "Age18_59", "Age60_above"]
+men_means = [87.0, 232, 541, 63]
+women_means = [72, 226, 550, 70]
+width = 0.35  # the width of the bars: can also be len(x) sequence
+fig, ax = plt.subplots()
+ax.bar(labels, men_means, width, label="Male")
+ax.bar(labels, women_means, width, bottom=men_means, label="Female")
+ax.set_ylabel("Number of beneficiaries")
+ax.set_title("Beneficiaries by age group and gender")
+ax.legend()
+plt.savefig(f"{IMAGE_DIR}/beneficiaries_by_age_groups.png")
+plt.show()
+
+# %%
+# % of household members age 60 and above
+age_60 = (
+    beneficiaries_by_age_group["female60+"] + beneficiaries_by_age_group["male60+"]
+) / sum(beneficiaries_by_age_group.values())
+logging.info(age_60)
+
+# %%
+# % of beneficiaries age under 5
+age_under5 = (
+    beneficiaries_by_age_group["female0_5"] + beneficiaries_by_age_group["male0_5"]
+) / sum(beneficiaries_by_age_group.values())
+logging.info(age_under5)
+
+# %%
+# visualizing how informants were notified about NRCS package distribution
+data_df[
+    "How were you notified about the relief delivery date"
+].value_counts().sort_values().plot(kind="barh")
+plt.title("How were you notified about the relief delivery date")
+plt.xlabel("Frequency")
+plt.savefig(f"{IMAGE_DIR}/notification_means.png")
+
+
+# %%
+# free text attribute found in the dataset
+len(
+    data_df[
+        "Are there any suggestion you want to give to Nepal Red Cross society to improve the relief distribution program in the future Mention if any"
+    ].unique()
+)
+
+# %% [markdown]
+# # Some thoughts arising from the data
+# 1. Which communication means adopted by nepal RC proved more effective for the affected communities?
+# 2. Of the financial aid provided, how can one prioritize the needs of the communities based on how the aid was used?
+#
+# 3. based on the time taken to arrive at the relief distribution centers, how convenient is it to reduce the number of people who travelled between 3-4 hours to get the aid? Can new centers that would reduce this travel time be created?
+#
+# 4. For the communication modes used, were resources equitably distributed to the different channels?
+#
+
+# %%
+# replace the 9 null values with the mode for the specific column on mode of communication.
```

Comment:
Hi Pius I wanted to check what the reason was for filling the missing values in this column withe mode?

Also, I think you can do this by using the mode function in pandas eg:

df['col'] = df['col'].fillna(df['col'].mode()[0])

Reply:
Mode was used because we are dealing with communication method and I think a participant would have learned of the event through one of the available means. That is why I chose the most used means to fill the few nulls.